src/Microsoft.NetCore.Analyzers/Core/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -132,7 +134,9 @@ public override void Initialize(AnalysisContext analysisContext)
                     if (targetMethod.IsGenericMethod || targetMethod.ContainingType == null || targetMethod.ContainingType.IsErrorType() ||
                         (targetMethod.ContainingType != null &&
                          (activatorType != null && activatorType.Equals(targetMethod.ContainingType)) ||
-                         (resourceManagerType != null && resourceManagerType.Equals(targetMethod.ContainingType))))
+                         (resourceManagerType != null && resourceManagerType.Equals(targetMethod.ContainingType)) ||
+                         (charType != null && charType.Equals(targetMethod.ContainingType)) ||
```

Comment:
The if condition starts to be a bit ugly IMO maybe that would be worth creating some method, WDYT?

Reply:
That might be nice, but I don't think we need to make that change now. Keeping the current form isn't beautiful but I can certainly review it quickly. 😄